js/private/js_binary.sh.tpl

```diff
@@ -195,22 +195,42 @@ if [ ! -f "$entry_point" ]; then
     exit 1
 fi
 
-export JS_BINARY__NODE_BINARY="$JS_BINARY__RUNFILES/{{workspace_name}}/{{node}}"
-if [ ! -f "$JS_BINARY__NODE_BINARY" ]; then
-    logf_fatal "node binary '%s' not found in runfiles" "$JS_BINARY__NODE_BINARY"
-    exit 1
+node="$(_normalize_path "{{node}}")"
+if [ "${node:0:1}" = "/" ]; then
+    # A user may specify an absolute path to node using target_tool_path in node_toolchain
+    export JS_BINARY__NODE_BINARY="$node"
+    if [ ! -f "$JS_BINARY__NODE_BINARY" ]; then
+        logf_fatal "node binary '%s' not found" "$JS_BINARY__NODE_BINARY"
+        exit 1
+    fi
+else
+    export JS_BINARY__NODE_BINARY="$JS_BINARY__RUNFILES/{{workspace_name}}/{{node}}"
+    if [ ! -f "$JS_BINARY__NODE_BINARY" ]; then
+        logf_fatal "node binary '%s' not found in runfiles" "$JS_BINARY__NODE_BINARY"
+        exit 1
+    fi
 fi
 if [ "$_IS_WINDOWS" -ne "1" ] && [ ! -x "$JS_BINARY__NODE_BINARY" ]; then
     logf_fatal "node binary '%s' is not executable" "$JS_BINARY__NODE_BINARY"
     exit 1
 fi
 
-npm={{npm}}
+npm="{{npm}}"
```

Comment:
Why does node need the `_normalize_path` but not npm?

Reply:
the both have it, I just check that npm is not empty before I call it a few lines down